docs/promo-codes/index.md

```diff
@@ -0,0 +1,462 @@
+# Promo Codes
+
+As International Burger Day looms ever closer, Madame Jellobutter is eager to
+start her burger-related promotions. She bought ads on local billboards and she
+even put on a giant burger costume to pass flyers at a local music festival[^1].
+Depending on the ad, potential customers will either see the promo code FREE,
+which corresponds to the "buy n burgers, get n/2 burgers free" discount, or
+HALF, which corresponds to the "buy a burger with everything on it and get half
+off the entire order" discount.
+
+## `getDiscountFunction` function
+
+Add a new function that maps a given promo code to its corresponding discount
+function:
+
+<<< Discount.re#simple-get-discount
+
+Instead of
+[Js.String.toUpperCase](https://melange.re/v3.0.0/api/re/melange/Js/String/#val-toLowerCase),
+we could've used
+[String.uppercase_ascii](https://melange.re/v3.0.0/api/re/melange/Stdlib/String/#val-uppercase_ascii),
+but its name implies, `String.uppercase_ascii` can only handle strings
```

Comment:
In this case, because the function matches ascii strings, we could have used it right?